account_invoice_tax_note/models/account_invoice.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def _get_account_tax_groups(self):
+        """
+        Get every account.tax.group used on current recordset where
+        report_note is filled.
+        Force the company into context if invoices are not in the user company
+        :return: account.tax.group recordset
+        """
+        invoices = self
+        company = invoices.mapped("company_id")
+        company.ensure_one()
+        if company.id != self.env.user.company_id.id:
+            # Update the context only if it's necessary
+            invoices = invoices.with_context(force_company=company.id)
```

Comment:
I suggest `self.ensure_one()` and not testing for the company at all (or maybe there is a use case for using this method on a recordset that I don't understand).